### PR TITLE
catalog-import: Improve how URLs are analyzed for add/import

### LINKED ISSUE #3752: Catalago registration using repo on azure, does not fill in the filepathtype variable
When trying to pass the link of the repo pointing to the file catalog-info.yaml or template.yaml, the code link does not fill in the variable filepathtype of the parseGitUri library, thus requesting a pull request to add a catalog file.
the link used in the test was [https://dev.azure.com/DEVOPS-RCHLO/cell-cac/_git/cac.backstage.custom?path=%2Fcatalog-info.yaml&version=GBmaster&_a=contents](url)

image

## Expected Behavior

Detect the presence of the .yaml file and skip the step of creating a pull request

## Steps to Reproduce

1. Step 1
2. Step 2
3. ...

## Your Environment

- NodeJS Version (v12):
- v14.15.0
- Operating System and Version (e.g. Ubuntu 14.04):
- Windows 10 pro 10.0.18363
- Browser Information:
O Microsoft Edge  87.0.664.60

### PR DESCRIPTION
**Pull request by freben**
Fixes #3752

### CODE REVIEW

**Comment by freben** [comment]
@dtuite @Xantier maybe have a look at this one if it makes sense?

**Comment by freben** [comment]
I know that this is a very minimal fix, and there's other things around it that could be improved (exercising all these places with tests, for one). But just getting the immediate bug out of the way for now.

**Comment by Xantier** [comment]
> @dtuite @Xantier maybe have a look at this one if it makes sense?

This will cover the immediate problem at least. I can't think of a case/repo URL that would contain `.yaml`/`.yml` (without the file itself being referenced) for now so it should be alrighty.

File: plugins/catalog-import/src/util/urls.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import parseGitUri from 'git-url-parse';
+
+export type UrlType = 'file' | 'repo';
```
Comment (Rugvip) [inline plugins/catalog-import/src/util/urls.ts:19]: `'file'` should be `'blob'` right?

What does `'repo'` actually represent? 😅

Comment (freben) [reply plugins/catalog-import/src/util/urls.ts:19]: We can just choose some convention - sure, `blob` is what the underlying lib uses.

The semantics of the other one is unclear still. The plugin is under development and this is subject to change. The original code looked at the url type to see if the user had pasted exactly a repo url (no filepath at all under it), and then triggering the import code flow. But the code actually should support both a repo root and a folder within a repo.

So essentially ... best effort, to be iterated on.

Comment (Rugvip) [reply plugins/catalog-import/src/util/urls.ts:19]: Still a bit confused though, is `'repo'` actually a possible output of `parseGitUri`?

Comment (freben) [reply plugins/catalog-import/src/util/urls.ts:19]: Supposedly, yes.

Comment (freben) [reply plugins/catalog-import/src/util/urls.ts:19]: `The type of filepath in the url ("blob" or "tree").`

https://github.com/IonicaBizau/git-url-parse

But it's all so sloppy in this lib - it's sometimes filled in, sometimes not, sometimes things are suddenly the empty string. So the code I added was looking for only the exact match of `'repo'` and nothing else.

**Review by Rugvip — APPROVED** [review]

**Pull request merged**